pkg/scheduler/dependency_graph.go

```diff
@@ -338,29 +368,44 @@ func (rl *ReplicaList) Swap(i, j int) {
 	rl.Items[i], rl.Items[j] = rl.Items[j], rl.Items[i]
 }
 
-func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]string, error) {
-	label := labels.Set{"flow": flow.Name}
+func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]client.Replica, []client.Replica, error) {
+	name := flow.Name
+	if options.FlowInstanceName != "" {
+		name = options.FlowInstanceName
+	}
+	label := labels.Set{"flow": name}
 	var tmp ReplicaList
```

Comment:
Not sure I got your question right. ReplicaList != client.ReplicaList. The purpose of tmp is to convert existingReplicas to the ReplicaList type which is based on client.ReplicaList type and, in addition, implements 3 additional methods that are required to sort the collection. In other words this is a way to sort replicas by custom criteria